api/listhandler.py

```diff
@@ -0,0 +1,239 @@
+# @author:  Renzo Frigato
+
+import datetime
+import logging
+import base
+import json
+import util
+import copy
+import os
+
+from . import permchecker
+from . import files
+
+log = logging.getLogger('scitran.api')
+
+
+class ListHandler(base.RequestHandler):
+    """
+    This class handle operations on a generic sublist of a container like tags, group roles, user permissions, etc.
+
+    The pattern used is:
+    1) initialize request
+    2) exec request
+    3) check and return result
+
+    Specific behaviors (permissions checking logic for authenticated and not superuser users, storage interaction)
+    are specified in the routes defined in api.py
+    """
+
+    def __init__(self, request=None, response=None):
+        super(ListHandler, self).__init__(request, response)
+        self._initialized = None
+
+    def get(self, *args, **kwargs):
```

Comment:
remove permcheck and storage from kwargs and indicate them explicitly
remove *args from the function signature